src/hotspot/share/opto/type.cpp

```diff
@@ -836,32 +976,27 @@ const Type *Type::meet_helper(const Type *t, bool include_speculative) const {
     return result->make_narrowklass();
   }
 
+#ifdef ASSERT
+  Compile* C = Compile::current();
+  VerifyMeetMark verif(C);
+#endif
+
   const Type *this_t = maybe_remove_speculative(include_speculative);
   t = t->maybe_remove_speculative(include_speculative);
 
   const Type *mt = this_t->xmeet(t);
 #ifdef ASSERT
-  if (isa_narrowoop() || t->isa_narrowoop()) return mt;
-  if (isa_narrowklass() || t->isa_narrowklass()) return mt;
-  Compile* C = Compile::current();
-  if (!C->_type_verify_symmetry) {
+  VerifyMeetResult* type_verify = C->_type_verify;
```

Comment:
As a style nit, you could access `VerifyMeetResult*` through `VerifyMeetMark` or even encapsulate it there.

Reply:
I assume you suggested to hide access  to cache (`add()`, `meet()`) in `VerifyMeetMark`. I agree. `VerifyMeetMark` can be renamed `VerifyMeet` in such case.